pkg/endpoint/endpoint.go

```diff
@@ -366,6 +370,11 @@ func (e *Endpoint) UpdateController(name string, params controller.ControllerPar
 	e.controllers.UpdateController(name, params)
 }
 
+// TriggerController triggers the endpoint's controller with the specified name.
+func (e *Endpoint) TriggerController(name string) {
```

Comment:
nit: I think this function does not necessarily need to be exported in the current implementation. We could even consider inlining it at the call site.